profiling/coverage.md

```diff
@@ -0,0 +1,217 @@
+  that help to improve code coverage
+
+![Coverage Analysis](./coverage-screenshot.png)
+
+The individual steps of instrumenting code with [BisectPPX] are greatly
+abstracted by OCamlfind (OCaml's library manager) and OCamlbuild
+(OCaml's compilation manager):
+
+    # write code
+    vim example.ml
+
+    # build it with instrumentation from bisect_ppx
+    ocamlbuild -use-ocamlfind -pkg bisect_ppx -pkg unix example.native
+    
+    # execute it - generates files ./bisect*.out
+    ./example.native
+    
+    # generate report
+    bisect-ppx-report -I _build -html coverage bisect000*
+    
+    # view coverage/index.html
+
+    Summary:
+     - 'binding' points: 2/2 (100.00%)
+     - 'sequence' points: 10/10 (100.00%)
+     - 'match/function' points: 5/8 (62.50%)
+     - total: 17/20 (85.00%)
+
+The fourth step generates a HTML report in `coverage/`. All it takes is
+to declare to [OCamlbuild] that a module depends on `bisect_ppx` and it
+will be instrumented during compilation. Behind the scenes `ocamlfind`
+makes sure that the compiler uses a preprocessing step that instruments
+the code.
+
+## Signal Handling 
+
+During execution the code instrumentation leads to the collection of
+data. This code registers a function with `at_exit` that writes the data
+to `bisect*.out` when `exit` is called. A binary can terminate without
+calling `exit` and in that case the file would not be written. It is
+therefore important to make sure that `exit` is called. If this does not
+happen naturally, for example in the context of a daemon that is
+terminated by receiving the `TERM` signal, a signal handler must be
+installed:
+
+    let stop signal =
+      printf "caught signal %d\n" signal;
+      exit 0
+
+    Sys.set_signal Sys.sigterm (Sys.Signal_handle stop)
+
+## Where Data is Written
+
+By default, [BisectPPX] writes data in a binary's current working
+directory as `bisectXXXX.out`. It doesn't overwrite existing files and
+files from several runs can be combined during analysis. However, this
+name and the location can be inconvenient when multiple programs share a
+directory.
+
+[BisectPPX]'s default can be overridden with the `BISECT_FILE`
+environment variable. This can happen on the command line:
+
+    BISECT_FILE=/tmp/example ./example.native
+
+In the context of XenServer we could do this in startup scripts. 
+However, we added a bit of code 
+
+    val Coverage.init: string -> unit
+
+that sets the environment variable from inside the program. The files
+are written to a temporary directory (respecting `$TMP` or using `/tmp`)
+and uses the `string`-typed argument to include it in the name. To be
+effective, this function must be called before the programs exits. For
+clarity it is called at the begin of program execution.
+
+## Instrumenting an Oasis Project
+
+While instrumentation is easy on the level of a small file or project it
+is increasingly less so in a bigger project. We decided to focus on
```

Comment:
'increasingly less so' is a bit awkward. Perhaps 'it becomes more challenging as the project grows'?